great_expectations/self_check/util.py

```diff
@@ -385,7 +385,7 @@ def get_dataset(
         return PandasDataset(df, profiler=profiler, caching=caching)
 
     elif dataset_type == "sqlite":
-        if not create_engine:
+        if not create_engine or not SQLITE_TYPES:
```

Comment:
It scares me that we need to make these kinds of changes.

This reinforces my view that we should not be relying on imports to determine which backends to test against. Instead, I would strongly advocate for an explicit config file for this.

Reply:
When imports fail for those dialects, we are setting the `*_TYPES` dicts to `{}`, but later in this function we are just trying to access them with square bracket notation (not `.get()`). There were already bailout points at `if not create_engine` and it thought it made sense to bail out at that point if those dicts were empty